src/openfl/_internal/text/TextEngine.hx

```diff
@@ -1012,3 +1019,37 @@ class TextEngine {
 		return text;
 	}
 }
+
+// this is a helper class for dealing with word separators
+// it's forced to be inline (so always stack-allocated)
+private class WordSeparator {
+	public var index(default,null):Int;
+	public var render(default,null):Bool;
+
+	public extern inline function new(index, render) {
+		this.index = index;
+		this.render = render;
+	}
+
+	public extern inline function setFrom(other:WordSeparator) {
+		this.index = other.index;
+		this.render = other.render;
+	}
+
+	public static extern inline function find(text:String, startIndex:Int):WordSeparator {
+		var index = -1;
+		var render = false;
+		var space = text.indexOf(" ", startIndex);
+		var hyphenRegExp = ~/(?<=\S)-(?=\S)/;
```

Comment:
hmm, could this be done without regexp? or at least without lookahead/lookbehinds? I'm not sure right now but I think this is rather expensive and IIRC the text engine is already quite laggy.

Reply:
Don't think it's possible without regexp. 
eg. "red-blue" is a word-breaking hyphen "-10" is not. But the `~/\b\-\b/` I replaced it with should be easier on the perfomance.